### ISSUE TITLE
Cannot specify region/zone in timestamp

### ISSUE THREAD

**Issue by sam701** [fluent/fluentd#2454]
- fluentd or td-agent version: 1.4.2
- Environment information: docker image `fluentd:v1.4-2`

#### configuration

```
<source>
  @type dummy
  dummy {"time":"6/12/19 14:30:25:052", "message": "hello"}
  tag tail
</source>

<filter tail>
  @type parser
  key_name time
  reserve_data true

  <parse>
    @type regexp
    expression (?<time>.*)
    time_format %m/%d/%y %H:%M:%S:%N
    timezone Europe/Berlin
  </parse>
</filter>

<match **>
  @type stdout
</match>

```

#### Problem
According to the [documentation](https://docs.fluentd.org/configuration/parse-section), it is possible to specify "Region/Zone" as timezone: `timezone Europe/Berlin`. But `fluentd` fails to start with the following message:
```
fluentd -c /fluentd/etc/fluentd-client.conf
2019-06-12 15:38:43 +0000 [info]: parsing config file is succeeded path="/fluentd/etc/fluentd-client.conf"
/usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/time.rb:206:in `-': nil can't be coerced into Integer (TypeError)
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/time.rb:206:in `initialize'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/time.rb:169:in `new'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/time.rb:169:in `time_parser_create'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/plugin/parser.rb:64:in `configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/plugin/parser_regexp.rb:34:in `configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/plugin.rb:164:in `configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/plugin_helper/parser.rb:90:in `block in configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/plugin_helper/parser.rb:85:in `each'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/plugin_helper/parser.rb:85:in `configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/plugin/filter_parser.rb:51:in `configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/plugin.rb:164:in `configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/agent.rb:152:in `add_filter'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/agent.rb:70:in `block in configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/agent.rb:64:in `each'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/agent.rb:64:in `configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/root_agent.rb:150:in `configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/engine.rb:131:in `configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/engine.rb:96:in `run_configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/supervisor.rb:801:in `run_configure'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/supervisor.rb:579:in `dry_run'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/supervisor.rb:597:in `supervise'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/supervisor.rb:502:in `run_supervisor'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/lib/fluent/command/fluentd.rb:310:in `<top (required)>'
	from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	from /usr/lib/ruby/gems/2.5.0/gems/fluentd-1.4.2/bin/fluentd:8:in `<top (required)>'
	from /usr/bin/fluentd:23:in `load'
	from /usr/bin/fluentd:23:in `<main>'
```

It works though when I specify the timezone like `timezone +0200`.

**Comment by repeatedly** [comment]
https://github.com/fluent/fluentd/pull/2421 ?

**Comment by sam701** [comment]
@repeatedly thanks, will this fix be available in some stable release soon?

**Comment by ganmacs** [comment]
it is already available on v1.5.1 
https://github.com/fluent/fluentd/blob/12ffa0afe313737bf09900efce8c8656baaf77bc/CHANGELOG.md#bug-fixes
https://github.com/fluent/fluentd/releases/tag/v1.5.1
